src/OutputWriters/netcdf_output_writer.jl

```diff
@@ -173,91 +757,135 @@ function add_schedule_metadata!(global_attributes, schedule::AveragedTimeInterva
     return nothing
 end
 
+#####
+##### NetCDFOutputWriter definition and constructor
+#####
+
+mutable struct NetCDFOutputWriter{G, D, O, T, A, FS, DN} <: AbstractOutputWriter
+    grid :: G
+    filepath :: String
+    dataset :: D
+    outputs :: O
+    schedule :: T
+    array_type :: A
+    indices :: Tuple
+    global_attributes :: Dict
+    output_attributes :: Dict
+    dimensions :: Dict
+    with_halos :: Bool
+    include_grid_metrics :: Bool
+    overwrite_existing :: Bool
+    verbose :: Bool
+    deflatelevel :: Int
+    part :: Int
+    file_splitting :: FS
+    dimension_name_generator :: DN
+end
+
 """
-    NetCDFOutputWriter(model, outputs; filename, schedule,
+    NetCDFOutputWriter(model, outputs;
+                       filename,
+                       schedule,
                        grid = model.grid,
                        dir = ".",
                        array_type = Array{Float64},
-                       indices = nothing,
-                       with_halos = false,
+                       indices = (:, :, :),
                        global_attributes = Dict(),
                        output_attributes = Dict(),
                        dimensions = Dict(),
-                       overwrite_existing = false,
+                       with_halos = false,
+                       include_grid_metrics = true,
+                       overwrite_existing = nothing,
+                       verbose = false,
                        deflatelevel = 0,
                        part = 1,
                        file_splitting = NoFileSplitting(),
-                       verbose = false)
+                       dimension_name_generator = default_dim_name)
```

Comment:
This is a great way to do things. Maybe in the future we can include functions that already follow a given popular/useful convention. I think we should probably also provide one that uses the exact names Oceananigans uses (I assume that'll be needed for the integration with `FieldTimeSeries`?)

Reply:
Yeah I didn't want to dig into standards/conventions for this PR, but I'm hoping a custom `dimension_name_generator` would be enough to produce output that conforms to certain conventions. If not, we can extend it!

I didn't make the default use the Oceananigans names as it included unicode and superscripts, which I like and Python 3 supports, but then users may struggle to work with the data efficiently outside of Julia.

Not sure if using the Oceananigans names would make it much easier to integrate with `FieldTimeSeries`, but if it does then I'll definitely add such a `dimension_name_generator`.

Do you think it's necessary for this PR or should we wait for the subsequent PR to support `FieldTimeSeries` before deciding?